### PR TITLE
Fix #1088 set schema on the TypedWrite

### LINKED ISSUE #1088: IllegalArgumentException when writing avro using AvroIO.TypedWrite
```
[info]   java.lang.IllegalArgumentException: Unless using DynamicDestinations, .withSchema() is required.
[info]   at org.apache.beam.sdk.repackaged.com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
[info]   at org.apache.beam.sdk.io.AvroIO$TypedWrite.expand(AvroIO.java:1173)
[info]   at org.apache.beam.sdk.io.AvroIO$TypedWrite.expand(AvroIO.java:848)
[info]   at org.apache.beam.sdk.Pipeline.applyInternal(Pipeline.java:537)
[info]   at org.apache.beam.sdk.Pipeline.applyTransform(Pipeline.java:491)
[info]   at org.apache.beam.sdk.values.PCollection.apply(PCollection.java:299)
[info]   at com.spotify.scio.values.PCollectionWrapper.applyInternal(PCollectionWrapper.scala:44)
[info]   at com.spotify.scio.values.PCollectionWrapper.applyInternal$(PCollectionWrapper.scala:42)
[info]   at com.spotify.scio.values.SCollectionImpl.applyInternal(SCollection.scala:1279)
[info]   at com.spotify.scio.values.SCollection.saveAsTypedAvroFile(SCollection.scala:976)
```

### PR DESCRIPTION
**Pull request by andrewsmartin**

### CODE REVIEW

File: scio-avro/src/main/scala/com/spotify/scio/avro/types/AvroType.scala [hunk 1]
```diff
@@ -231,7 +231,7 @@ object AvroType {
  *
  * This decouples generated fields and methods from macro expansion to keep core macro free.
  */
-class AvroType[T: TypeTag] {
+class AvroType[T: TypeTag] extends Serializable {
```
Comment (nevillelyh) [inline scio-avro/src/main/scala/com/spotify/scio/avro/types/AvroType.scala]: Where is `AvroType` pulled into a lambda?

File: scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * under the License.
+ */
+
+package com.spotify.scio.avro
+
+import com.google.common.io.Files
+import com.spotify.scio.{ContextAndArgs, ScioResult}
+import com.spotify.scio.testing.PipelineSpec
+
+class TypedAvroIT extends PipelineSpec {
```
Comment (nevillelyh) [inline scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: Don't really need to extend `PipelineSpec` right? Not even using any matchers so `extends FlatSpec` should be enough.

OTOH, is there anything we should assert after running the job?

File: scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  "Typed Avro IO" should "not throw exception" in {
+    val tempDir = Files.createTempDir()
+    TypedAvroJob.run(tempDir.getAbsolutePath)
+    tempDir.deleteOnExit()
+  }
+}
+
+object TypedAvroJob {
+  def run(tempDir: String): ScioResult = {
```
Comment (nevillelyh) [inline scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: For the sake of consistency (and less confusion to those new to the code base), use the standard `def main():Unit = ...`? You can pass in a `--output=$tempDir` arg.

File: scio-avro/src/main/scala/com/spotify/scio/avro/types/AvroType.scala (hunk 1 above)
Comment (andrewsmartin) [reply scio-avro/src/main/scala/com/spotify/scio/avro/types/AvroType.scala]: In the function that converts typed objects to generic record. https://github.com/spotify/scio/pull/1089/files#diff-417755bd55a6dbb7a50a9f58d6a7af4bL973

File: scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala (hunk 2 above)
Comment (andrewsmartin) [reply scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: 👍 can change to just stick with `FlatSpec`. I thought about asserting that the file contents are as expected but it seemed a bit out of scope for this test...happy to include something like that if you think it's worthwhile though. I mainly wrote the test to make sure the fix works.

Comment (nevillelyh) [reply scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: Maybe use the `noException should be thrownBy` matcher then?
http://www.scalatest.org/user_guide/using_matchers

Comment (andrewsmartin) [reply scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: 👍

File: scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala (hunk 3 above)
Comment (andrewsmartin) [reply scio-test/src/it/scala/com/spotify/scio/avro/TypedAvroIT.scala]: 👍

**Pull request merged**